azcopy/pathUtils.go

```diff
@@ -69,3 +71,82 @@ func GetContainerName(path string, location common.Location) (string, error) {
 		return "", fmt.Errorf("cannot get container name on location type %s", location.String())
 	}
 }
+
+// ----- LOCATION LEVEL HANDLING -----
+type LocationLevel uint8
+
+var ELocationLevel LocationLevel = 0
+
+func (LocationLevel) Account() LocationLevel   { return 0 } // Account is never used in AzCopy, but is in testing to understand resource management.
+func (LocationLevel) Service() LocationLevel   { return 1 }
+func (LocationLevel) Container() LocationLevel { return 2 }
+func (LocationLevel) Object() LocationLevel    { return 3 } // An Object can be a directory or object.
+
+// Uses syntax to assume the "level" of a location.
+// This is typically used to
+func DetermineLocationLevel(location string, locationType common.Location, source bool) (LocationLevel, error) {
```

Comment:
A thought about types that both azcopy and cmd potentially need and thus should be exported-- Is it feasible to migrate them to another common package, i.e. (`github.com/Azure/azure-storage-azcopy/azcopy/common`), that way both packages can utilize them, but they are not potentially confusing a customer?